mitosheet/mitosheet/tests/updates/test_update_existing_import.py

```diff
@@ -0,0 +1,327 @@
+                        'error_bad_lines': [False],
+                    }
+                }
+           ]
+        }
+    ]
+
+    mito.update_existing_imports(updated_import_obj)
+    mito.undo()
+
+    # Make sure the updates occured correctly 
+    assert mito.dfs[0].equals(pd.DataFrame({'Unnamed: 0': [0, 1, 2], 'A': [1, 2, 3], 'B': [1, 2, 3]}, index=[0, 1, 2]))
+
+
+def test_redo_works():
+    # Make dataframes and files for test
+    df1 = pd.DataFrame(data={'A': [1, 2, 3]})
+    df1.to_csv(TEST_CSV_FILE, index=True)
+    df2 = pd.DataFrame(data={'A': [10, 20, 30]})
+    df2.to_csv(TEST_CSV_FILE_TWO, index=True)
+
+    # Create with no dataframes
+    mito = create_mito_wrapper_dfs()
+    # And then import just a test file
+    mito.simple_import([TEST_CSV_FILE])
+    step_id = mito.curr_step.step_id
+
+    mito.set_formula('=A', 0, 'B', True)
+
+    # Update the imports 
+    updated_import_obj = [ 
+        {
+           'step_id': step_id,
+           'imports': [
+                {
+                    'step_type': 'simple_import',
+                    'params': {
+                        'file_names': [TEST_CSV_FILE_TWO],
+                        'delimeters': [','],
+                        'encodings': ['utf-8'],
+                        'error_bad_lines': [False],
+                    }
+                }
+           ]
+        }
+    ]
+
+    mito.update_existing_imports(updated_import_obj)
+    mito.redo()
+
+    # Make sure the updates occured correctly 
+    assert mito.dfs[0].equals(pd.DataFrame({'Unnamed: 0': [0, 1, 2], 'A': [1, 2, 3], 'B': [1, 2, 3]}, index=[0, 1, 2]))
+
+
+def test_update_imports_is_atomic():
+    # Make dataframes and files for test
+    df = pd.DataFrame(data={'A': [1, 2, 3], 'B': [2, 3, 4]})
+    df1 = pd.DataFrame(data={'C': [1, 2, 3], 'D': [2, 3, 4]})
+    df1.to_csv(TEST_CSV_FILE, index=False, sep=',', encoding='utf-8')
+    with pd.ExcelWriter(TEST_EXCEL_FILE) as writer:  
+        df.to_excel(writer, sheet_name='Sheet1', index=False)
+    
+    # Create with no dataframes
+    mito = create_mito_wrapper_dfs()
+    mito.excel_import(TEST_EXCEL_FILE, ['Sheet1'], True, 0)
+    mito.simple_import([TEST_CSV_FILE])
+
+    mito.update_existing_imports([
+        {
+            'step_id': mito.steps_including_skipped[1].step_id,
+            'imports': [{
+                'step_type': 'simple_import',
+                'params': {
+                    'file_names': [TEST_CSV_FILE]
+                }
+            }]
+        },
+        {
+            'step_id': mito.steps_including_skipped[2].step_id,
+            'imports': [{
+                'step_type': 'simple_import',
+                'params': {
+                    'file_names': ['no such file']
+                }
+            }]
+        }
+    ])
+
+    assert mito.dfs[0].equals(df)
+    assert mito.dfs[1].equals(df1)
+
+
+def test_test_import_returns_good_data():
+    df = pd.DataFrame(data={'A': [1, 2, 3], 'B': [2, 3, 4]})
```

Comment:
TODO: Add tests for:
1. Multiple imports in the middle, still gets the correct indexes on the errors
2. Updating imports with multiple from the same step id, also works